src/serve_rendered.js

```diff
@@ -1045,16 +1045,40 @@ export const serve_rendered = {
       return false;
     }
 
-    if (styleJSON.sprite && !httpTester.test(styleJSON.sprite)) {
-      styleJSON.sprite =
-        'sprites://' +
-        styleJSON.sprite
-          .replace('{style}', path.basename(styleFile, '.json'))
-          .replace(
-            '{styleJsonFolder}',
-            path.relative(options.paths.sprites, path.dirname(styleJSONPath)),
-          );
+    if (styleJSON.sprite) {
+      if (Array.isArray(styleJSON.sprite)) {
+        styleJSON.sprite.forEach((spriteItem) => {
+          if (!httpTester.test(spriteItem.url)) {
+            spriteItem.url =
```

Comment:
I do agree this looks better and is simpler. My only concern my previous way preserved the existing sprite format, so if you were not using multi-sprites it would function as before and work with older versions of mapbox/maplibre that do not support multi-sprite.

Where this way basically forces multi-sprites on the users style whether they are using them or not. So it would only work with newer maplibre-js that supports that.

Reply:
I'm on the fence about reverting this change. I don't know if I want to break clients that doesn't support multi-sprite if the person making the style hasn't chosen to use them.